### PR TITLE
Deprecation of `azurerm.kubernetes_cluster.private_link_enabled` for `enable_private_cluster`

### LINKED ISSUE #6416: Support private cluster option for AKS clusters
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

This issue #2241 was closed due to the AKS API not supporting private clusters. Now that AKS support for private clusters is GA, please add this important feature. Private cluster feature is GA and documented in the API  https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/createorupdate 
Please note that this feature has a relationship with `api_server_authorized_ip_ranges` as only either can be enabled. 
### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "cluster" {
  enable_private_cluster = true ##if true then api_server_authorized_ip_ranges should be "false"
  ...
}
```

### References

* #2241

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #6416 

Motivation in the issue.

### CODE REVIEW

**Comment by aristosvo** [comment]
```
TF_ACC=1 go test -v ./azurerm/internal/services/containers/tests/ -run=AzureRMKubernetesCluster_privateCluster -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceAzureRMKubernetesCluster_privateCluster
--- PASS: TestAccDataSourceAzureRMKubernetesCluster_privateCluster (1302.93s)
=== RUN   TestAccAzureRMKubernetesCluster_privateClusterOn
=== PAUSE TestAccAzureRMKubernetesCluster_privateClusterOn
=== RUN   TestAccAzureRMKubernetesCluster_privateClusterOff
=== PAUSE TestAccAzureRMKubernetesCluster_privateClusterOff
=== CONT  TestAccAzureRMKubernetesCluster_privateClusterOn
=== CONT  TestAccAzureRMKubernetesCluster_privateClusterOff
--- PASS: TestAccAzureRMKubernetesCluster_privateClusterOff (1175.41s)
--- PASS: TestAccAzureRMKubernetesCluster_privateClusterOn (1228.63s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/tests    2531.589s
```

File: azurerm/internal/services/containers/data_source_kubernetes_cluster.go [hunk 1]
```diff
@@ -212,8 +212,18 @@ func dataSourceArmKubernetesCluster() *schema.Resource {
...
-				Type:     schema.TypeBool,
-				Computed: true,
+				Type:          schema.TypeBool,
+				Computed:      true,
+				Optional:      true,
+				ConflictsWith: []string{"enable_private_cluster"},
+				Deprecated:    "Deprecated in favor of `enable_private_cluster`", // TODO -- remove this in next major version
+			},
+
+			"enable_private_cluster": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/data_source_kubernetes_cluster.go:222]: could we be more consistent with the rest of the provider and name this
```suggestion
			"private_cluster_enabled": {
```

Comment (aristosvo) [reply azurerm/internal/services/containers/data_source_kubernetes_cluster.go]: Thanks! Fixed as suggested

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**